lib/Core/BuildEngine.cpp

```diff
@@ -77,7 +77,10 @@ class BuildEngineImpl : public BuildDBDelegate {
   std::unique_ptr<BuildEngineTrace> trace;
 
   /// The current build iteration, used to sequentially timestamp build results.
-  uint64_t currentTimestamp = 0;
+  Iteration currentIteration = 0;
+  
+  /// The timestamp when the build started or 0 if not running currently
+  Clock::Timestamp buildStartedTimestamp = 0;
```

Comment:
uint64 max probably a safer sentinel value, if we need a sentinel. If we switch to double, maybe we can use -1 and enforce negative values can never be output from Clock().

Reply:
I changed `Timestamp` to be of type `std::chrono::time_point` which has a static `max()` method. I think that's best as a sentinel value. In the `durationSince()` method I'm now checking for positive durations and will add that to the documentation.

Reply:
If using actual timestamps and not relative start and duration this property is not needed anymore.